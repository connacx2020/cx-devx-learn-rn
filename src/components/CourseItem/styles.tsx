import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container:{
        borderRadius: 5,
        width: 320,
        elevation: 5,
        backgroundColor: '#fff',
        shadowOffset: { width: 2, height: 1 },
        shadowColor: '#333',
        shadowOpacity: 0.5,
        shadowRadius: 5,
        marginHorizontal: 5,
        marginVertical: 10,
        padding: 1,
    },
    content : {

    },
    img: {
        shadowOffset: { width: 2, height: 1 },
        shadowColor: '#333',
        shadowOpacity: 0.5,
        shadowRadius: 5,
        borderColor:'#ededed',
        borderWidth:0.5,
        height:150,
        width:'100%',
        resizeMode: 'stretch'
    },
    title:{
        fontSize:18,
        fontStyle:'italic',
        paddingTop: 5,
        paddingLeft:5
    },
    footer:{
        flex:1,
        flexDirection:'row',
        padding:5
    },
    ratingContainer :{
       flexDirection:'row'
    },
    star:{
        color: "#FFD700"
    },
    ratingTxt : {
        marginTop:-2,
        fontStyle:'italic',
        fontSize:15,
        marginHorizontal:20
    },
    likes :{
        flexGrow:1,
        fontStyle:'italic',
        marginTop:-3,
        marginHorizontal:15,
        fontSize:15,
        textAlign:'right'
    }
});
