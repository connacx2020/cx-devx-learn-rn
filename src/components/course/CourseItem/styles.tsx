import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        borderRadius: 5,
        width: 320,
        elevation: 5,
        backgroundColor: '#fff',
        shadowOffset: { width: 2, height: 1 },
        shadowColor: '#333',
        shadowOpacity: 0.5,
        shadowRadius: 5,
        marginHorizontal: 5,
        marginVertical: 10,
        padding: 1
    },
    content: {
    },
    content_bottom: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        padding: 5
    },
    img_blank: {
        height: 150,
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        backgroundColor: '#2289f0'
    },
    img_blank_text: {
        alignSelf: 'center',
        fontWeight: 'bold',
        textAlign: 'center',
        fontSize: 30,
        color: 'white'
    },
    img: {
        shadowOffset: { width: 2, height: 1 },
        shadowColor: '#333',
        shadowOpacity: 0.5,
        shadowRadius: 5,
        borderColor: '#ededed',
        borderWidth: 0.5,
        height: 150,
        width: '100%',
        resizeMode: 'contain'
    },
    title: {
        fontSize: 18,
        fontStyle: 'italic',
        paddingLeft: 5,
        fontWeight: 'bold',
        alignSelf: 'center',
        flexShrink: 1,
        marginRight: 10
    },
    footer: {
        flex: 1,
        flexDirection: 'row',
        paddingBottom: 5,
        paddingLeft: 5,
        justifyContent: 'space-between'
    },
    ratingContainer: {
        flexDirection: 'row'
    },
    price_text: {
        paddingHorizontal: 10,
        alignSelf: 'center',
        fontSize: 15, fontWeight: 'bold'
    },
    star: {
        color: "#FFD700",
    },
    likes: {
        flexGrow: 1,
        marginTop: -3,
        marginHorizontal: 10,
        fontSize: 15,
        textAlign: 'right',
    }
});
