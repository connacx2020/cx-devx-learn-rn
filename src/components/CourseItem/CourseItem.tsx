import React from 'react';
import { TouchableOpacity, View, Text, Image, StyleProp, ViewStyle } from 'react-native';
import StarRating from 'react-native-star-rating';
import { useTheme } from '@react-navigation/native';

import { styles } from './styles';
import { Course } from '../../models';


export const CxDevxCourseItem:React.FC<any> = ({id, authorID, enrolled, img, title, rating, description, routeToCourseDetail}: any, ) => {
    const { colors } = useTheme();

    return (
        <View style={[styles.container, { backgroundColor: colors.background }]}>
            <View style={styles.content}>
                <TouchableOpacity onPress={() => routeToCourseDetail(id)}>
                    {
                        img === "" ? <View  style={{height: 150, display:'flex',flexDirection:'column',justifyContent:'center', backgroundColor:'#2289f0'}}><Text style={{alignSelf:'center', fontWeight:'bold', fontSize: 30, color:'white'}}>{title}</Text></View> :
                        <Image
                            testID="imgID"
                            style={styles.img}
                            source={{
                                uri: img
                            }} />
                    }
                    <Text testID="titleID" style={[styles.title, { color: colors.text }]}>{title}</Text>
                </TouchableOpacity>
            </View>
            <View style={styles.footer}>
                <View style={styles.ratingContainer} >
                    <StarRating
                        disabled={true}
                        maxStars={5}
                        rating={rating}
                        starSize={20}
                        starStyle={styles.star as StyleProp<ViewStyle>}
                    />
                </View>
                <Text style={[styles.likes, { color: colors.text }]}>{enrolled} Enrolls</Text>
            </View>


        </View>

    )

}
