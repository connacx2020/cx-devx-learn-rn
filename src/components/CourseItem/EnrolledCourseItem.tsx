import React from 'react';
import { View, Text, Image, StyleProp, ViewStyle } from 'react-native';
import StarRating from 'react-native-star-rating';
import { useTheme } from '@react-navigation/native';


import { styles } from './styles';
import { Course } from '../../models';


export const CxDevxEntrolledCourseItem: React.FC<any> = ({ id, authorID, enrolled, img, title, rating, description }: any,) => {
    const { colors } = useTheme();

    return (
        <View style={[styles.container, { backgroundColor: colors.background }]}>
            <View style={styles.content}>
                <View>
                    {
                        img === "" ? <View style={styles.img_blank}><Text numberOfLines={2} style={styles.img_blank_text}>{title}</Text></View> :
                            <Image
                                testID="imgID"
                                style={styles.img}
                                source={{
                                    uri: img
                                }} />
                    }
                    <View style={styles.content_bottom}>
                        <Text testID="titleID" numberOfLines={1} style={[styles.title, { color: colors.text, alignSelf: 'flex-start' }]}>{title}</Text>
                    </View>
                </View>
            </View>
            <View style={styles.footer}>
                <View style={styles.ratingContainer} >
                    <StarRating
                        disabled={true}
                        maxStars={5}
                        rating={rating}
                        starSize={20}
                        starStyle={styles.star as StyleProp<ViewStyle>}
                    />
                </View>
                <Text style={{ color: colors.text, marginHorizontal: 8 }}>{rating}</Text>
                <Text style={[styles.likes, { color: colors.text, paddingRight: 5 }]}>Enrolls: {enrolled}</Text>
            </View>


        </View>

    )

}
