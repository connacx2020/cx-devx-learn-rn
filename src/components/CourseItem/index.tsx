import React from 'react';
import { TouchableOpacity, View, Text, Image ,StyleProp,ViewStyle} from 'react-native';
import StarRating from 'react-native-star-rating';

import { styles } from './styles';

type courseItem = {
    img: string,
    title: string,
    rate: number,
    likes : number
    starStyle: StyleProp<ViewStyle>
}

export function CxDevxCourseItem({ img,title,rate,likes }: courseItem) {
    return(
        <View style={styles.container}>
                <View style={styles.content}>
                    <Image
                        style={styles.img}
                        source={{
                            uri: img
                        }}/>
                    <Text style={styles.title}>{title}</Text>
                </View>
                <View style={styles.footer}>
                    <View style={styles.ratingContainer}>
                        <StarRating
                            disabled={true}
                            maxStars={5}
                            rating={rate}
                            starSize={20}
                            starStyle={styles.star}
                        />
                        <Text style={styles.ratingTxt}>{rate}</Text>
                    </View>
                    <Text style={styles.likes}>{likes} Likes</Text>
                </View>

        </View>

    )

}
