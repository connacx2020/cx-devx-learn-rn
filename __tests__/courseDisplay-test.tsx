import React from 'react';
import { render, fireEvent, wait, act } from '@testing-library/react-native';
import { CxDevxCourseItem } from '../src/components/course/CourseItem/CourseItem';
import CxDevxFeed from '../src/components/Feed/Feed';
import { string } from 'yup';


describe("Course's Item UI test", () => {
    let routeToCourseDetail = jest.fn();
    let courseItemWrapper = render(<CxDevxCourseItem img="image" title="Apollo Server Express Tutorial" likes={1500} rate={4} routeToCourseDetail={routeToCourseDetail} />);
    it('Must include Title, Image ,StarRating, Rate and Likes , routeFunction props', () => {

        expect(courseItemWrapper.getByTestId('titleID').props.children).toBe('Apollo Server Express Tutorial');
        expect(courseItemWrapper.getByTestId('likeID').props.children[0]).toEqual(1500);
        expect(courseItemWrapper.getByTestId('rateID').props.children).toEqual(4);
        expect(courseItemWrapper.getByTestId('imgID').props.source.uri).toEqual("image");
        expect(courseItemWrapper.getByTestId("star").props.rating).toEqual(4);
    });

});

describe("Course Display Page UI Testing", () => {
    const courseData = [
        {
            img: "https://miro.medium.com/max/1400/1*uvd7Z4npUG8qulaQLjHcZw.jpeg",
            title: "GraphQL Advanced Course",
            rate: 4.5,
            likes: 1000
        },
        {
            img: "https://miro.medium.com/max/2880/1*xcDT-neKHP7E3quS9n30gw.png",
            title: "React Hooks Course",
            rate: 5,
            likes: 1500
        },
        {
            img: "https://railsware.com/blog/wp-content/uploads/2018/09/2400%D1%851260-rw-blog-node-js.png",
            title: "Nodejs Advanced Course",
            rate: 4,
            likes: 784
        },
        {
            img: "https://i.ytimg.com/vi/OdU9H-_d14Y/maxresdefault.jpg",
            title: "React Native With Typescript",
            rate: 4,
            likes: 1000
        },
        {
            img: "https://dist.neo4j.com/wp-content/uploads/20170524234854/graph-ql-graph-database-neo4j-integration.png",
            title: "GraphQL with Neo4j  Course",
            rate: 5,
            likes: 1000
        }

    ];
    let navigation = jest.fn();
    let FeedWrapper = render(<CxDevxFeed navigation={navigation} />);
    // console.log(FeedWrapper.debug());
    expect(FeedWrapper.getByTestId('carouselID')).toBeDefined();
    expect(FeedWrapper.getByTestId('carouselID').props.data).toMatchSnapshot(courseData);
})
