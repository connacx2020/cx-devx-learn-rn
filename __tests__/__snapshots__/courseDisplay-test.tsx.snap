// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "0": Object {
    "img": "https://miro.medium.com/max/1400/1*uvd7Z4npUG8qulaQLjHcZw.jpeg",
    "likes": 1000,
    "rate": 4.5,
    "title": "GraphQL Advanced Course",
  },
  "1": Object {
    "img": "https://miro.medium.com/max/2880/1*xcDT-neKHP7E3quS9n30gw.png",
    "likes": 1500,
    "rate": 5,
    "title": "React Hooks Course",
  },
  "2": Object {
    "img": "https://railsware.com/blog/wp-content/uploads/2018/09/2400%D1%851260-rw-blog-node-js.png",
    "likes": 784,
    "rate": 4,
    "title": "Nodejs Advanced Course",
  },
  "3": Object {
    "img": "https://i.ytimg.com/vi/OdU9H-_d14Y/maxresdefault.jpg",
    "likes": 1000,
    "rate": 4,
    "title": "React Native With Typescript",
  },
  "4": Object {
    "img": "https://dist.neo4j.com/wp-content/uploads/20170524234854/graph-ql-graph-database-neo4j-integration.png",
    "likes": 1000,
    "rate": 5,
    "title": "GraphQL with Neo4j  Course",
  },
}
`;
